### DISCUSSION TITLE
Documentation request: How to specify an encryption key.

### DISCUSSION THREAD

**Discussion by jsdevtom** [qgustavor/mega#129] [General]
[Here](https://mega.js.org/docs/api#properties-1) it is said that:

> ### Properties
>
> * `name` - File name*
> * `attributes` - Object of attributes*
> * `key` - File key (buffer)*
> * ...
>
> Values marked with \* are `null` or `undefined` when an encryption key isn't specified.

But how do we "specify an encryption key" and where?

**Comment by jsdevtom** [comment]
Turns out I had to use the folder owner's account. Not the account with whom it's shared.

**Comment by qgustavor** [comment]
The idea behind that note is the following, take the shared file example from the homepage:

```javascript
import { File } from 'megajs'
const file = File.fromURL('https://mega.nz/file/example#example')
await file.loadAttributes()
console.log(file.name) // file name
console.log(file.size) // file size in bytes
```

If you pass to `File.fromURL` a URL containing the file key then encrypted attributes such as the file name will be decrypted. The key in the URL is the part after the hash: `https://mega.nz/file/example#THIS_PART`. If you pass something like this `https://mega.nz/file/example` instead, without the file key, you can still access some metadata from the file but not encrypted contents.

**Reply by qgustavor** [reply]
By the way, files loaded using `Storage` always have their keys defined.  The only way not to define the file key is using `File`. Why it was implemented? Privacy: it allows making applications that can handle MEGA files without having to access encrypted contents, like a tool that notifies when there are new files in a shared folder. I made one that generated a RSS feed based on a shared folder a long time ago.

**Reply by qgustavor** [reply]
Documentation updated to be more clear about this. Discussion locked.
